avalanche/benchmarks/scenarios/continuous_task_agnostic_scenario.py

```diff
@@ -0,0 +1,330 @@
+################################################################################
+# Copyright (c) 2022 ContinualAI.                                              #
+# Copyrights licensed under the MIT License.                                   #
+# See the accompanying LICENSE file for terms.                                 #
+#                                                                              #
+# Date: 04-08-2023                                                             #
+# Author(s): Antonio Carta, Hamed Hemati                                       #
+# E-mail: contact@continualai.org                                              #
+# Website: avalanche.continualai.org                                           #
+################################################################################
+
+#  Code adapted from: https://github.com/chenzeno/FOO-VB/blob/main/datasets.py
+
+from functools import partial
+from typing import (
+
+    Iterable,
+    List,
+    Optional,
+    Union,
+)
+from typing_extensions import Literal
+
+import torch
+from torch.distributions.categorical import Categorical
+from torch.utils.data import Sampler
+
+from avalanche.benchmarks.scenarios.benchmark_wrapper_utils import wrap_stream
+from avalanche.benchmarks.utils import AvalancheDataset
+from avalanche.benchmarks.scenarios.generic_scenario import (
+    CLStream,
+    CLScenario,
+    DatasetExperience,
+    CLScenario,
+)
+from avalanche.benchmarks.utils.classification_dataset import (
+    classification_subset,
+    concat_classification_datasets
+)
+from avalanche.benchmarks.scenarios.online_scenario import (
+    OnlineClassificationExperience,
+    TClassificationDataset, TCLDataset
+)
+
+
+class CyclicSampler(Sampler):
+    """Samples elements from [0,..,len(dataset)-1] in a cyclic manner."""
+
+    def __init__(self, n_samples, shuffle=True, rng=None):
+        self.n_samples = n_samples
+        self.rng = rng
+        self.shuffle = shuffle
+        self._reset_indices()
+
+    def _reset_indices(self):
+        self.indices = torch.arange(self.n_samples).tolist()
+        if self.shuffle:
+            self.indices = torch.randperm(self.n_samples,
+                                          generator=self.rng).tolist()
+
+    def __iter__(self):
+        while True:
+            for idx in self.indices:
+                yield idx
+            self._reset_indices()
+
+    def __len__(self):
+        return self.n_samples
+
+
+def create_sub_exp_from_multi_exps(
+    online_benchmark: "OnlineCLScenario",
+    original_stream: Iterable[DatasetExperience[TClassificationDataset]],
```

Comment:
can we remove original_stream from the arguments? original streams should already be in `online_benchmark` anyway...